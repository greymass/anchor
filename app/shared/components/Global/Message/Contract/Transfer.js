// @flow
import React, { Component } from 'react';
import { Header, List } from 'semantic-ui-react';
import { translate } from 'react-i18next';

import GlobalMessageContract from './Base';

export class GlobalMessageContractTransfer extends Component<Props> {
  render() {
    const { data, t } = this.props;
    return (
      <GlobalMessageContract>
        <Header>
          <p>
            {t('system_transfer_title')}
          </p>
        </Header>
        <p>{t('system_transfer_1', data)}</p>
        <List style={{ marginLeft: '2em' }}>
          <List.Item><p>{t('system_transfer_2', data)}</p></List.Item>
          <List.Item><p>{t('system_transfer_3', data)}</p></List.Item>
          <List.Item><p>{t('system_transfer_4', data)}</p></List.Item>
        </List>
        <p>{t('system_transfer_5', data)}</p>
        <p>{t('system_transfer_6', data)}</p>
      </GlobalMessageContract>
    );
  }
}

export default translate('contracts')(GlobalMessageContractTransfer);
