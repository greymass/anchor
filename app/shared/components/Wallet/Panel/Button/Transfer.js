// @flow
import React, { Component } from 'react';
import { translate } from 'react-i18next';

import GlobalTransactionModal from '../../../Global/Transaction/Modal';
import WalletPanelFormTransfer from '../Form/Transfer';

type Props = {
  actions: {
    clearSystemState: () => void
  },
  balances: {},
  settings: {},
  system: {},
  t: {}
};

class WalletPanelButtonTransfer extends Component<Props> {
  props: Props;

  render() {
    const {
      actions,
      balances,
      settings,
      system,
      t
    } = this.props;
    return (
      <GlobalTransactionModal
        actionName="TRANSFER"
        actions={actions}
        button={{
          color: 'blue',
          content: t('transfer_button_cta'),
          icon: 'exchange'
        }}
        content={(
          <WalletPanelFormTransfer
            actions={actions}
            balances={balances}
            settings={settings}
            system={system}
          />
        )}
        icon="exchange"
        title={t('transfer_modal_title')}
        settings={settings}
        system={system}
      />
    );
  }
}

export default translate('transfer')(WalletPanelButtonTransfer);
