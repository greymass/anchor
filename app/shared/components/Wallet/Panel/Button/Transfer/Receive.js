// @flow
import React, { Component } from 'react';
import { translate } from 'react-i18next';
import { Button } from 'semantic-ui-react';

import WalletPanelModalTransferReceive from '../../Modal/Transfer/Receive';

class WalletPanelButtonTransferReceive extends Component<Props> {
  state = {
    open: false
  }

  onOpen = () => this.setState({ open: true });

  onClose = () => this.setState({ open: false });

  render() {
    const {
      accountName,
      t,
    } = this.props;

    const {
      open
    } = this.state;

    return (
      <WalletPanelModalTransferReceive
        accountName={accountName}
        fluid
        onClose={this.onClose}
        open={open}
        trigger={(
          <Button
            color="blue"
            content={t('transfer_receive_button_cta')}
            fluid
            icon="arrow circle down"
            onClick={this.onOpen}
          />
        )}
      />
    );
  }
}

export default translate('transfer')(WalletPanelButtonTransferReceive);
