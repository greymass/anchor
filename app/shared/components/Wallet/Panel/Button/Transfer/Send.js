// @flow
import React, { Component } from 'react';
import { translate } from 'react-i18next';

import GlobalTransactionModal from '../../../../Global/Transaction/Modal';
import WalletPanelFormTransfer from '../../Form/Transfer/Send';

type Props = {
  actions: {
    clearSystemState: () => void
  },
  balances: {},
  settings: {},
  system: {},
  t: () => void
};

class WalletPanelButtonTransfer extends Component<Props> {
  props: Props;

  render() {
    const {
      actions,
      balances,
      settings,
      system,
      t
    } = this.props;
    return (
      <GlobalTransactionModal
        actionName="TRANSFER"
        actions={actions}
        button={{
          color: 'blue',
          content: t('transfer_send_button_cta'),
          icon: 'arrow circle up'
        }}
        content={(
          <WalletPanelFormTransfer
            actions={actions}
            balances={balances}
            settings={settings}
            system={system}
          />
        )}
        icon="arrow circle up"
        title={t('transfer_modal_title')}
        settings={settings}
        system={system}
      />
    );
  }
}

export default translate('transfer')(WalletPanelButtonTransfer);
