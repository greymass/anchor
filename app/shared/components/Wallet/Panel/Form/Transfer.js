 // @flow
import React, { Component } from 'react';
import { Button, Divider, Form, Header, Icon, Message, Segment, Table } from 'semantic-ui-react';
import { I18n } from 'react-i18next';

import FormFieldAccount from '../../../Global/Form/Field/Account';
import FormFieldGeneric from '../../../Global/Form/Field/Generic';
import FormFieldMultiToken from '../../../Global/Form/Field/MultiToken';

export default class WalletPanelFormTransfer extends Component<Props> {
  constructor(props) {
    super(props);
    this.state = {
      confirming: false,
      from: props.settings.account,
      memo: '',
      quantity: '',
      to: '',
      waiting: false
    };
  }

  state = {};

  onChange = (e, { name, value }) => {
    this.setState({ [name]: value });
  }

  onKeyPress = (e) => {
    if (e.key === 'Enter') {
      this.onSubmit(e);
    }
  }

  onSubmit = () => {
    this.setState({
      confirming: true,
      waiting: true
    });
    // Make the user wait 3 seconds before they can confirm
    setTimeout(() => {
      this.setState({
        waiting: false
      });
    }, 3000);
  }

  onCancel = (e) => {
    this.setState({
      confirming: false,
      waiting: false
    });
    e.preventDefault();
    return false;
  }

  onConfirm = (e) => {
    const {
      from,
      to,
      quantity,
      memo
    } = this.state;
    this.setState({ confirming: false }, () => {
      this.props.actions.transfer(from, to, quantity, memo);
    });
    e.preventDefault();
    return false;
  }

  render() {
    const {
      balances,
      onClose,
      settings,
      system
    } = this.props;
    const {
      confirming,
      from,
      memo,
      quantity,
      to,
      waiting
    } = this.state;
    const balance = balances[settings.account];
    const asset = "EOS";
    const error = system.TRANSFER_LAST_ERROR;
    const errorMsg = (error && error.error) ? error.error.details[0].message : JSON.stringify(error);
    return (
      <I18n ns="transfer">
        {
          (t) => (
            <Form
              loading={system.TRANSFER === 'PENDING'}
              onKeyPress={this.onKeyPress}
              onSubmit={this.onSubmit}
            >
              {(confirming)
                ? (
                  <Segment basic clearing vertical>
                    <Header size="small">
                      {t('transfer_confirming_title')}
                      <Header.Subheader>
                        {t('transfer_confirming_body')}
                      </Header.Subheader>
                    </Header>
                    <Table compact definition striped>
                      <Table.Body>
                        <Table.Row>
                          <Table.Cell width={4}>{t('transfer_label_from')}</Table.Cell>
                          <Table.Cell>{from}</Table.Cell>
                        </Table.Row>
                        <Table.Row>
                          <Table.Cell>{t('transfer_label_to')}</Table.Cell>
                          <Table.Cell>{to}</Table.Cell>
                        </Table.Row>
                        <Table.Row>
                          <Table.Cell>{t('transfer_label_quantity')}</Table.Cell>
                          <Table.Cell>{quantity}</Table.Cell>
                        </Table.Row>
                        <Table.Row>
                          <Table.Cell>{t('transfer_label_memo')}</Table.Cell>
                          <Table.Cell>{memo}</Table.Cell>
                        </Table.Row>
                      </Table.Body>
                    </Table>
                    <Message info>
                      <Header>
                        <Icon name="warning sign" />
                        <Header.Content>
                          {t('transfer_warning_irreversible_body')}
                        </Header.Content>
                      </Header>
                    </Message>
                    <Divider />
                    <Button
                      onClick={this.onCancel}
                    >
                      <Icon name="x" /> {t('cancel')}
                    </Button>
                    <Button
                      color={(waiting) ? 'grey' : 'green'}
                      content={(waiting) ? t('waiting') : t('confirm')}
                      disabled={waiting}
                      floated="right"
                      onClick={this.onConfirm}
                    />
                  </Segment>
                )
                : (
                  <Segment basic clearing>
                    <FormFieldAccount
                      disabled
                      label={t('transfer_label_from')}
                      name="from"
                      onChange={this.onChange}
                      value={settings.account}
                    />
                    <FormFieldAccount
                      autoFocus
                      label={t('transfer_label_to')}
                      name="to"
                      onChange={this.onChange}
                      value={to}
                    />
                    <FormFieldMultiToken
                      assets={Object.keys(balances[settings.account])}
                      icon="x"
                      label={t('transfer_label_token_and_quantity')}
                      loading={false}
                      maximum={balance[asset]}
                      name="quantity"
                      onChange={this.onChange}
                      value={quantity}
                    />
                    <FormFieldGeneric
                      icon="x"
                      label={t('transfer_label_memo')}
                      loading={false}
                      name="memo"
                      onChange={this.onChange}
                      value={memo}
                    />
                    {(error)
                      ? (
                        <Message negative>
                          {errorMsg}
                        </Message>
                      )
                      : ''
                    }
                    <Divider />
                    <Button
                      onClick={onClose}
                    >
                      <Icon name="x" /> {t('cancel')}
                    </Button>
                    <Button
                      content={t('confirm')}
                      floated="right"
                      primary
                    />
                  </Segment>
                )
              }
            </Form>
          )
        }
      </I18n>
    );
  }
}
