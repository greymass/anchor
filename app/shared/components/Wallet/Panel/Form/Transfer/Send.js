// @flow
import React, { Component } from 'react';
import { Button, Divider, Form, Message, Icon, Segment, Table } from 'semantic-ui-react';
import { withTranslation } from 'react-i18next';
import { findIndex, isObject } from 'lodash';
import { get } from 'dot-prop-immutable';
import debounce from 'lodash/debounce';
import { Asset } from '@greymass/eosio';

import { createHttpHandler } from '../../../../../utils/http/handler';
import FormFieldMultiToken from '../../../../Global/Form/Field/MultiToken';
import FormMessageError from '../../../../Global/Form/Message/Error';
import GlobalFormFieldAccount from '../../../../Global/Form/Field/Account';
import GlobalFormFieldKeyPublic from '../../../../Global/Form/Field/Key/Public';
import GlobalFormFieldMemo from '../../../../Global/Form/Field/Memo';
import WalletPanelFormTransferSendConfirming from './Send/Confirming';

const initialState = {
  confirming: false,
  fioFee: false,
  formError: false,
  formId: false,
  memo: '',
  memoValid: true,
  quantity: '',
  quantitySet: false,
  submitDisabled: true,
  to: '',
  toValid: true,
  waiting: false,
  waitingStarted: 0
};

class WalletPanelFormTransferSend extends Component<Props> {
  constructor(props) {
    super(props);
    this.state = Object.assign({}, initialState, {
      asset: props.connection.chainSymbol || 'EOS',
      from: props.settings.account,
    });
  }

  componentWillReceiveProps(nextProps) {
    const {
      balances,
      connection,
      system,
      wallet
    } = nextProps;
    const { TRANSFER_SET_ASSET_DATA } = system;
    if (this.state.from !== nextProps.settings.account) {
      this.setState({ from: nextProps.settings.account });
    }
    if (
      connection.chainSymbol === 'FIO'
      && (
        this.props.connection.chainSymbol !== 'FIO'
        || this.state.fioFee === false
      )
    ) {
      this.setState({ fioFee: 'loading' }, () => {
        this.loadFIOFee();
      });
    }
    if (TRANSFER_SET_ASSET_DATA) {
      const { amount, asset } = TRANSFER_SET_ASSET_DATA;
      const assetContract = balances.__contracts[asset.toUpperCase()];
      let precision;
      if (assetContract) {
        ({ precision } = assetContract);
      }
      let precisionValue = connection.tokenPrecision;
      if (precision) {
        if (isObject(precision)) {
          precisionValue = precision[asset.toUpperCase()];
        }
        if (precision > 0) {
          precisionValue = precision;
        }
      }
      const quantity = `${amount.toFixed(precisionValue)} ${asset}`;
      if (
        this.state.asset !== TRANSFER_SET_ASSET_DATA.asset
        || this.state.quantity !== quantity
      ) {
        this.setState({
          asset: TRANSFER_SET_ASSET_DATA.asset,
          quantity,
          quantitySet: Date.now(),
        }, () => {
          this.onChange(null, {
            name: 'quantity',
            value: quantity,
            valid: true,
          });
        });
      }
    }
  }

  onConfirm = () => {
    const {
      from,
      memo,
      quantity,
      asset,
      to
    } = this.state;
    this.setState({ confirming: false }, () => {
      this.props.actions.transfer(from, to, quantity, memo, asset);
    });
  };

  onSubmit = () => {
    this.setState({
      confirming: true,
      waiting: true,
      waitingStarted: new Date()
    });
    this.interval = setInterval(this.tick, 250);
    if (this.props.isConfirming) {
      this.props.isConfirming(true);
    }
    // Make the user wait 3 seconds before they can confirm
    setTimeout(() => {
      clearInterval(this.interval);
      this.setState({
        waiting: false
      });
    }, 3000);
  };

  componentWillUnmount() {
    clearInterval(this.interval);
  }

  tick = () => this.setState({ waiting: true });

  onCancel = (e) => {
    this.setState(Object.assign({}, initialState, {
      formId: new Date()
    }));
    if (this.props.onClose) {
      this.props.onClose();
    }
    if (this.props.isConfirming) {
      this.props.isConfirming(false);
    }
    e.preventDefault();
    return false;
  };

  getContractHash = debounce((value) => {
    const { actions } = this.props;

    actions.getContractHash(value);
  }, 400);

  onChange = (e, { name, value, valid }) => {
    this.props.actions.clearSystemState();
    if (name === 'to') {
      const {
        settings
      } = this.props;
      const {
        contacts
      } = settings;

      const position = findIndex(contacts, { accountName: value });

      if (position > -1) {
        this.onChange(e, { name: 'memo', value: contacts[position].defaultMemo || '', valid: true });
      }

      this.getContractHash(value);
    }

    const newState = { [name]: value };
    if (name === 'quantity') {
      const [, asset] = value.split(' ');
      newState.asset = asset;
    }

    newState[`${name}Valid`] = valid;

    newState.submitDisabled = false;
    newState.formError = false;

    this.setState(newState, () => {
      const error = this.errorInForm();

      if (error) {
        this.onError(error);
      }
    });
  };

  onError = (error) => {
    let formError;

    if (error !== true) {
      formError = error;
    }

    this.setState({
      formError,
      submitDisabled: true
    });
  };

  onBack = () => {
    if (this.props.isConfirming) {
      this.props.isConfirming(false);
    }
    this.setState({
      confirming: false
    });
  };

  errorInForm = () => {
    const {
      memo,
      memoValid,
      quantity,
      to,
      toValid
    } = this.state;

    const {
      app,
      settings
    } = this.props;

    if (!to || to === '') {
      return true;
    }

    if (!quantity || quantity === '') {
      return true;
    }

    if (!toValid) {
      return 'invalid_accountName';
    }

    if (!memoValid) {
      return 'invalid_memo';
    }

    if (to === settings.account) {
      return 'cannot_transfer_to_self';
    }
    const exchangeAccounts = get(app, 'constants.exchanges') || [];
    if (exchangeAccounts.includes(to) && (!memo || memo.length === 0)) {
      return 'transferring_to_exchange_without_memo';
    }

    return false;
  };

  fillMax = () => {
    const {
      balances,
      connection,
      settings,
    } = this.props;
    const {
      asset,
      fioFee,
    } = this.state;
    const balance = balances[settings.account];
    const { precision } = balances.__contracts[asset.toUpperCase()];
    let precisionValue = connection.tokenPrecision;
    if (precision) {
      if (isObject(precision)) {
        precisionValue = precision[asset.toUpperCase()];
      }
      if (precision > 0) {
        precisionValue = precision;
      }
    }
    const quantity = Asset.from(balance[asset], `${precisionValue},${asset}`);
    if (fioFee) {
      quantity.value -= fioFee / 1000000000;
    }
    this.setState({
      quantity: String(quantity),
      quantitySet: Date.now(),
    }, () => {
      this.onChange(null, {
        name: 'quantity',
        value: String(quantity),
        valid: true,
      });
    });
  }

  loadFIOFee = async () => {
    const { connection, wallet } = this.props;
    const { httpClient } = await createHttpHandler(connection);
    const fees = await httpClient.post(`${connection.httpEndpoint}/v1/chain/get_fee`, {
      end_point: 'transfer_tokens_pub_key',
      fio_address: wallet.address,
    });
    const { fee } = fees.data;
    this.setState({ fioFee: fee });
  }

  render() {
    const {
      app,
      balances,
      connection,
      settings,
      system,
      t
    } = this.props;
    const {
      asset,
      confirming,
      fioFee,
      formError,
      formId,
      from,
      memo,
      quantity,
      quantitySet,
      submitDisabled,
      to,
      waiting,
      waitingStarted
    } = this.state;

    const balance = balances[settings.account];
    const assetContract = balances.__contracts[asset.toUpperCase()];
    let precision = connection.tokenPrecision;
    if (assetContract && assetContract.precision && assetContract.precision[asset]) {
      precision = assetContract.precision[asset];
    }

    const quantityValue = quantity ? parseFloat(quantity.split(' ')[0]) : 0;

    if (!balance) return false;

    let exchangeWarning;

    const exchangeAccounts = get(app, 'constants.exchanges') || [];

    if (memo && memo !== '' && exchangeAccounts && exchangeAccounts[connection.chainId]) {
      exchangeAccounts[connection.chainId].forEach((exchangeAccount) => {
        if (memo.match(`.*?${exchangeAccount}.*?`)) {
          exchangeWarning = (
            <Message warning>
              {`${t('transfer_send_exchange_in_memo_one')} ${exchangeAccount} ${t('transfer_send_exchange_in_memo_two')}`}
            </Message>
          );
        }
      });
    }

    const shouldDisplayTransferingToContractMessage =
      to &&
      system.ACCOUNT_HAS_CONTRACT_LAST_ACCOUNT === to &&
      system.ACCOUNT_HAS_CONTRACT_LAST_CONTRACT_HASH &&
      system.ACCOUNT_HAS_CONTRACT_LAST_CONTRACT_HASH !== '0000000000000000000000000000000000000000000000000000000000000000';

    const hasWarnings = exchangeWarning || shouldDisplayTransferingToContractMessage;

    return (
      <Form
        key={formId}
        loading={system.TRANSFER === 'PENDING'}
        onKeyPress={this.onKeyPress}
        onSubmit={this.onSubmit}
        warning={hasWarnings}
      >
        {(confirming)
          ? (
            <WalletPanelFormTransferSendConfirming
              asset={asset}
              balances={balances}
              from={from}
              memo={memo}
              onBack={this.onBack}
              onConfirm={this.onConfirm}
              quantity={quantity}
              to={to}
              waiting={waiting}
              waitingStarted={waitingStarted}
            />
          ) : (
            <Segment basic clearing>
              {(connection.keyPrefix === 'FIO')
                ? (
                  <GlobalFormFieldKeyPublic
                    app={app}
                    autoFocus
                    connection={connection}
                    contacts={settings.contacts}
                    enableContacts
                    enableExchanges
                    chainId={connection.chainId}
                    fluid
                    label={t('transfer_label_to_pubkey')}
                    name="to"
                    onChange={this.onChange}
                    value={to}
                  />
                )
                : (
                  <GlobalFormFieldAccount
                    app={app}
                    autoFocus
                    contacts={settings.contacts}
                    enableContacts={false}
                    enableExchanges={false}
                    chainId={connection.chainId}
                    fluid
                    label={t('transfer_label_to')}
                    name="to"
                    onChange={this.onChange}
                    value={to}
                  />
                )
              }
              {(shouldDisplayTransferingToContractMessage) && (
                <Message
                  content={t('transfer_destination_account_is_contract')}
                  icon="warning sign"
                  warning
                />
              )}
              <FormFieldMultiToken
                balances={balances}
                connection={connection}
                icon="x"
                label={(
                  <span>
                    <span style={{ float: 'right' }}>
                      <a
                        onClick={this.fillMax}
                        style={{ cursor: 'pointer' }}
                      >
                        {String(Asset.from(balance[asset], `${precision},${asset}`))}
                      </a>
                    </span>
                    {t('transfer_label_token_and_quantity')}
                  </span>
                )}
                key={quantitySet}
                loading={false}
                maximum={balance[asset]}
                name="quantity"
                onChange={this.onChange}
                settings={settings}
                value={quantity}
              />
              {(connection.keyPrefix === 'FIO')
                ? (
                  <Table size="small">
                    <Table.Row>
                      <Table.Cell textAlign="right">Tokens to Send</Table.Cell>
                      <Table.Cell>{quantity || '0'}</Table.Cell>
                    </Table.Row>
                    <Table.Row>
                      <Table.Cell textAlign="right">Transaction Fee</Table.Cell>
                      <Table.Cell>{fioFee / 1000000000} FIO</Table.Cell>
                    </Table.Row>
                    <Table.Row>
                      <Table.Cell textAlign="right">Total</Table.Cell>
                      <Table.Cell>
                        {(quantityValue + (fioFee / 1000000000)).toFixed(9)} FIO
                      </Table.Cell>
                    </Table.Row>
                  </Table>
                )
                : false
              }
              <GlobalFormFieldMemo
                icon="x"
                label={t('transfer_label_memo')}
                loading={false}
                name="memo"
                onChange={this.onChange}
                value={memo}
              />

              <FormMessageError
                error={formError}
              />

              { exchangeWarning }

              <Segment basic>
                <Button
                  content={t('confirm')}
                  disabled={submitDisabled}
                  floated="right"
                  primary
                />
                <Button
                  onClick={this.onCancel}
                >
                  <Icon name="undo" /> {t('reset')}
                </Button>

              </Segment>
            </Segment>
          )}
      </Form>
    );
  }
}

export default withTranslation('transfer')(WalletPanelFormTransferSend);
