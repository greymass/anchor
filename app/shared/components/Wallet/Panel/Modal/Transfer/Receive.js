// @flow
import React, { Component } from 'react';
import { translate } from 'react-i18next';
import { Button, Segment, Header, Icon, Modal } from 'semantic-ui-react';

class WalletPanelModalTransferReceive extends Component<Props> {
  render() {
    const {
      accountName,
      onClose,
      open,
      t,
      trigger
    } = this.props;

    return (
      <Modal
        centered={false}
        trigger={trigger}
        onClose={onClose}
        open={open}
        size="tiny"
      >
        <Header icon="arrow circle down" content={t('transfer_receive_modal_title')} />
        <Modal.Content>
          <Segment basic clearing vertical textAlign="center">
            <Header>
              {t('transfer_receive_body')}
            </Header>
            <Segment
              basic
              clearing
              textAlign="center"
              style={{ padding: '2rem' }}
            >
              <h1>{accountName}</h1>
            </Segment>
          </Segment>
        </Modal.Content>
        <Modal.Actions>
          <Button
            onClick={onClose}
          >
            <Icon name="x" /> {t('cancel')}
          </Button>
        </Modal.Actions>
      </Modal>
    );
  }
}

export default translate('transfer')(WalletPanelModalTransferReceive);
