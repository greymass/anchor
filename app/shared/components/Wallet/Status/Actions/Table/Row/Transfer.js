// @flow
import React, { Component } from 'react';
import { translate } from 'react-i18next';
import { Icon } from 'semantic-ui-react';

class WalletStatusActionsTableRowTransfer extends Component<Props> {
  render() {
    const {
      data,
      t
    } = this.props;

    const iconStyle = {
      margin: '5px',
      marginRight: '20px',
      float: 'left',
      verticalAlign: 'top'
    };

    return (
      <div>
        <Icon
          name="arrow circle up"
          size="large"
          style={iconStyle}
        />
        <div>
          {`${t('actions_table_row_transfer_text_one')} ${data.quantity} ${t('actions_table_row_transfer_text_two')} ${data.to}.`}
        </div>
      </div>
    );
  }
}

export default translate('actions')(WalletStatusActionsTableRowTransfer);

