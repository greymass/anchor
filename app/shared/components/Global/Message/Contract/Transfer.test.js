import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { GlobalMessageContractTransfer } from './Transfer';

const mockProps = {
  children: <React.Fragment />,
  data: {},
  t: jest.fn,
};

describe('WalletMessageContractTransfer', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<GlobalMessageContractTransfer {...mockProps} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
