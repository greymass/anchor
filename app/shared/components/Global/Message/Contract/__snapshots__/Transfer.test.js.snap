// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletMessageContractTransfer renders correctly 1`] = `
<Translate(GlobalMessageContractBase)>
  <Header>
    <p>
      <Component />
    </p>
  </Header>
  <p>
    <Component />
  </p>
  <List
    style={
      Object {
        "marginLeft": "2em",
      }
    }
  >
    <ListItem>
      <p>
        <Component />
      </p>
    </ListItem>
    <ListItem>
      <p>
        <Component />
      </p>
    </ListItem>
    <ListItem>
      <p>
        <Component />
      </p>
    </ListItem>
  </List>
  <p>
    <Component />
  </p>
  <p>
    <Component />
  </p>
</Translate(GlobalMessageContractBase)>
`;
